{
	"document": {
		"aggregate_severity": {
			"namespace": "https://www.suse.com/support/security/rating/",
			"text": "moderate"
		},
		"category": "csaf_security_advisory",
		"csaf_version": "2.0",
		"distribution": {
			"text": "Copyright 2024 SUSE LLC. All rights reserved.",
			"tlp": {
				"label": "WHITE",
				"url": "https://www.first.org/tlp/"
			}
		},
		"lang": "en",
		"notes": [
			{
				"category": "summary",
				"text": "Security update for python-urllib3",
				"title": "Title of the patch"
			},
			{
				"category": "description",
				"text": "This update for python-urllib3 fixes the following issues:\n\npython-urllib3 was updated to version 1.22 (fate#326733, bsc#1110422) and contains new features and lots of bugfixes:\n\nThe full changelog can be found on:\n\n        https://github.com/Lukasa/urllib3/blob/1.22/CHANGES.rst\n\nSecurity issues fixed:\n\n- CVE-2016-9015: TLS certificate validation vulnerability (bsc#1024540).\n  (This issue did not affect our previous version 1.16.)\n\nNon security issues fixed:\n\n- bsc#1074247: Fix test suite, use correct date (gh#shazow/urllib3#1303).\n",
				"title": "Description of the patch"
			},
			{
				"category": "details",
				"text": "SUSE-2019-139,SUSE-OpenStack-Cloud-Magnum-Orchestration-7-2019-139,SUSE-SLE-Module-Public-Cloud-12-2019-139,SUSE-SUSE-Manager-Server-3.1-2019-139,SUSE-SUSE-Manager-Server-3.2-2019-139,SUSE-Storage-4-2019-139,SUSE-Storage-5-2019-139",
				"title": "Patchnames"
			},
			{
				"category": "legal_disclaimer",
				"text": "CSAF 2.0 data is provided by SUSE under the Creative Commons License 4.0 with Attribution (CC-BY-4.0).",
				"title": "Terms of use"
			}
		],
		"publisher": {
			"category": "vendor",
			"contact_details": "https://www.suse.com/support/security/contact/",
			"name": "SUSE Product Security Team",
			"namespace": "https://www.suse.com/"
		},
		"references": [
			{
				"category": "external",
				"summary": "SUSE ratings",
				"url": "https://www.suse.com/support/security/rating/"
			},
			{
				"category": "self",
				"summary": "URL of this CSAF notice",
				"url": "https://ftp.suse.com/pub/projects/security/csaf/suse-su-2019_0139-1.json"
			},
			{
				"category": "self",
				"summary": "URL for SUSE-SU-2019:0139-1",
				"url": "https://www.suse.com/support/update/announcement/2019/suse-su-20190139-1/"
			},
			{
				"category": "self",
				"summary": "E-Mail link for SUSE-SU-2019:0139-1",
				"url": "https://lists.suse.com/pipermail/sle-security-updates/2019-January/005056.html"
			},
			{
				"category": "self",
				"summary": "SUSE Bug 1024540",
				"url": "https://bugzilla.suse.com/1024540"
			},
			{
				"category": "self",
				"summary": "SUSE Bug 1074247",
				"url": "https://bugzilla.suse.com/1074247"
			},
			{
				"category": "self",
				"summary": "SUSE Bug 1110422",
				"url": "https://bugzilla.suse.com/1110422"
			},
			{
				"category": "self",
				"summary": "SUSE CVE CVE-2016-9015 page",
				"url": "https://www.suse.com/security/cve/CVE-2016-9015/"
			}
		],
		"title": "Security update for python-urllib3",
		"tracking": {
			"current_release_date": "2019-01-21T14:54:30Z",
			"generator": {
				"date": "2019-01-21T14:54:30Z",
				"engine": {
					"name": "cve-database.git:bin/generate-csaf.pl",
					"version": "1"
				}
			},
			"id": "SUSE-SU-2019:0139-1",
			"initial_release_date": "2019-01-21T14:54:30Z",
			"revision_history": [
				{
					"date": "2019-01-21T14:54:30Z",
					"number": "1",
					"summary": "Current version"
				}
			],
			"status": "final",
			"version": "1"
		}
	},
	"product_tree": {
		"branches": [
			{
				"branches": [
					{
						"branches": [
							{
								"category": "product_version",
								"name": "python-urllib3-1.22-3.10.1.noarch",
								"product": {
									"name": "python-urllib3-1.22-3.10.1.noarch",
									"product_id": "python-urllib3-1.22-3.10.1.noarch"
								}
							},
							{
								"category": "product_version",
								"name": "python3-urllib3-1.22-3.10.1.noarch",
								"product": {
									"name": "python3-urllib3-1.22-3.10.1.noarch",
									"product_id": "python3-urllib3-1.22-3.10.1.noarch"
								}
							}
						],
						"category": "architecture",
						"name": "noarch"
					},
					{
						"branches": [
							{
								"category": "product_name",
								"name": "SUSE Linux Enterprise Module for Public Cloud 12",
								"product": {
									"name": "SUSE Linux Enterprise Module for Public Cloud 12",
									"product_id": "SUSE Linux Enterprise Module for Public Cloud 12",
									"product_identification_helper": {
										"cpe": "cpe:/o:suse:sle-module-public-cloud:12"
									}
								}
							},
							{
								"category": "product_name",
								"name": "SUSE Manager Server 3.1",
								"product": {
									"name": "SUSE Manager Server 3.1",
									"product_id": "SUSE Manager Server 3.1",
									"product_identification_helper": {
										"cpe": "cpe:/o:suse:suse-manager-server:3.1"
									}
								}
							},
							{
								"category": "product_name",
								"name": "SUSE Manager Server 3.2",
								"product": {
									"name": "SUSE Manager Server 3.2",
									"product_id": "SUSE Manager Server 3.2",
									"product_identification_helper": {
										"cpe": "cpe:/o:suse:suse-manager-server:3.2"
									}
								}
							},
							{
								"category": "product_name",
								"name": "SUSE Enterprise Storage 4",
								"product": {
									"name": "SUSE Enterprise Storage 4",
									"product_id": "SUSE Enterprise Storage 4",
									"product_identification_helper": {
										"cpe": "cpe:/o:suse:ses:4"
									}
								}
							},
							{
								"category": "product_name",
								"name": "SUSE Enterprise Storage 5",
								"product": {
									"name": "SUSE Enterprise Storage 5",
									"product_id": "SUSE Enterprise Storage 5",
									"product_identification_helper": {
										"cpe": "cpe:/o:suse:ses:5"
									}
								}
							}
						],
						"category": "product_family",
						"name": "SUSE Linux Enterprise"
					}
				],
				"category": "vendor",
				"name": "SUSE"
			}
		],
		"relationships": [
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python-urllib3-1.22-3.10.1.noarch as component of SUSE Linux Enterprise Module for Public Cloud 12",
					"product_id": "SUSE Linux Enterprise Module for Public Cloud 12:python-urllib3-1.22-3.10.1.noarch"
				},
				"product_reference": "python-urllib3-1.22-3.10.1.noarch",
				"relates_to_product_reference": "SUSE Linux Enterprise Module for Public Cloud 12"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python3-urllib3-1.22-3.10.1.noarch as component of SUSE Linux Enterprise Module for Public Cloud 12",
					"product_id": "SUSE Linux Enterprise Module for Public Cloud 12:python3-urllib3-1.22-3.10.1.noarch"
				},
				"product_reference": "python3-urllib3-1.22-3.10.1.noarch",
				"relates_to_product_reference": "SUSE Linux Enterprise Module for Public Cloud 12"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python-urllib3-1.22-3.10.1.noarch as component of SUSE Manager Server 3.1",
					"product_id": "SUSE Manager Server 3.1:python-urllib3-1.22-3.10.1.noarch"
				},
				"product_reference": "python-urllib3-1.22-3.10.1.noarch",
				"relates_to_product_reference": "SUSE Manager Server 3.1"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python-urllib3-1.22-3.10.1.noarch as component of SUSE Manager Server 3.2",
					"product_id": "SUSE Manager Server 3.2:python-urllib3-1.22-3.10.1.noarch"
				},
				"product_reference": "python-urllib3-1.22-3.10.1.noarch",
				"relates_to_product_reference": "SUSE Manager Server 3.2"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python-urllib3-1.22-3.10.1.noarch as component of SUSE Enterprise Storage 4",
					"product_id": "SUSE Enterprise Storage 4:python-urllib3-1.22-3.10.1.noarch"
				},
				"product_reference": "python-urllib3-1.22-3.10.1.noarch",
				"relates_to_product_reference": "SUSE Enterprise Storage 4"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python-urllib3-1.22-3.10.1.noarch as component of SUSE Enterprise Storage 5",
					"product_id": "SUSE Enterprise Storage 5:python-urllib3-1.22-3.10.1.noarch"
				},
				"product_reference": "python-urllib3-1.22-3.10.1.noarch",
				"relates_to_product_reference": "SUSE Enterprise Storage 5"
			}
		]
	},
	"vulnerabilities": [
		{
			"cve": "CVE-2016-9015",
			"ids": [
				{
					"system_name": "SUSE CVE Page",
					"text": "https://www.suse.com/security/cve/CVE-2016-9015"
				}
			],
			"notes": [
				{
					"category": "general",
					"text": "Versions 1.17 and 1.18 of the Python urllib3 library suffer from a vulnerability that can cause them, in certain configurations, to not correctly validate TLS certificates. This places users of the library with those configurations at risk of man-in-the-middle and information leakage attacks. This vulnerability affects users using versions 1.17 and 1.18 of the urllib3 library, who are using the optional PyOpenSSL support for TLS instead of the regular standard library TLS backend, and who are using OpenSSL 1.1.0 via PyOpenSSL. This is an extremely uncommon configuration, so the security impact of this vulnerability is low.",
					"title": "CVE description"
				}
			],
			"product_status": {},
			"references": [
				{
					"category": "external",
					"summary": "CVE-2016-9015",
					"url": "https://www.suse.com/security/cve/CVE-2016-9015"
				},
				{
					"category": "external",
					"summary": "SUSE Bug 1023502 for CVE-2016-9015",
					"url": "https://bugzilla.suse.com/1023502"
				},
				{
					"category": "external",
					"summary": "SUSE Bug 1024540 for CVE-2016-9015",
					"url": "https://bugzilla.suse.com/1024540"
				}
			],
			"remediations": [
				{
					"category": "vendor_fix",
					"details": "To install this SUSE Security Update use the SUSE recommended installation methods like YaST online_update or \"zypper patch\".\n",
					"product_ids": [
						"SUSE Enterprise Storage 4:python-urllib3-1.22-3.10.1.noarch",
						"SUSE Enterprise Storage 5:python-urllib3-1.22-3.10.1.noarch",
						"SUSE Linux Enterprise Module for Public Cloud 12:python-urllib3-1.22-3.10.1.noarch",
						"SUSE Linux Enterprise Module for Public Cloud 12:python3-urllib3-1.22-3.10.1.noarch",
						"SUSE Manager Server 3.1:python-urllib3-1.22-3.10.1.noarch",
						"SUSE Manager Server 3.2:python-urllib3-1.22-3.10.1.noarch"
					]
				}
			],
			"scores": [
				{
					"cvss_v3": {
						"baseScore": 3.7,
						"baseSeverity": "LOW",
						"vectorString": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:L/I:N/A:N",
						"version": "3.1"
					},
					"products": [
						"SUSE Enterprise Storage 4:python-urllib3-1.22-3.10.1.noarch",
						"SUSE Enterprise Storage 5:python-urllib3-1.22-3.10.1.noarch",
						"SUSE Linux Enterprise Module for Public Cloud 12:python-urllib3-1.22-3.10.1.noarch",
						"SUSE Linux Enterprise Module for Public Cloud 12:python3-urllib3-1.22-3.10.1.noarch",
						"SUSE Manager Server 3.1:python-urllib3-1.22-3.10.1.noarch",
						"SUSE Manager Server 3.2:python-urllib3-1.22-3.10.1.noarch"
					]
				}
			],
			"threats": [
				{
					"category": "impact",
					"date": "2019-01-21T14:54:30Z",
					"details": "low"
				}
			],
			"title": "CVE-2016-9015"
		}
	]
}
